src/Symfony/Component/Notifier/Bridge/FreeMobile/FreeMobileTransport.php

```diff
@@ -63,7 +63,7 @@ protected function doSend(MessageInterface $message): SentMessage
             'json' => [
                 'user' => $this->login,
                 'pass' => $this->password,
-                'msg' => \utf8_encode($message->getSubject()),
+                'msg' => utf8_encode($message->getSubject()),
```

Comment:
This makes little sense to me: the subject is already UTF-8 encoded, so here we're doubly-encoding the UTF-8 string?
Do you have any link/doc/etc that this is correct?

Reply:
`utf8_encode` is really `convert_iso_1859_1_to_utf8` (there is even php.internals discussion about renaming the function to reflect that). If the input is already UTF-8, the output is corrupted garbage. So this change looks totally wrong to me.

Reply:
hi @nicolas-grekas 
Here is a screenshot of my phone message app, first message without and second with this PR code change:
![Uploading Capture d’écran . 2022-01-26 à 15.41.29.png…]()

my code is like this:

```php
// $texter Symfony/Component/Notifier/TexterInterface
$subject = ''; // this is the text I need with accents
$texter->send(new SmsMessage('06...', $subject));
```

Reply:
(please send the screenshot again)

Reply:
looks like you validated the comment form while the upload was still in progress...

Reply:
sent again my bad ^^

Reply:
also, which encoding are you using in your `$subject` variable ? Here, you show an empty string, which won't have encoding issues, so this is clearly not a reproducer.

Reply:
@stof I've written for testing `$subject = "àmÏg0";` here for my test before sending this


cf next code reproducer:

```php

// in app code
        $subject = 'àmÏg0';
        $this->texter->send(new SmsMessage('0661645818', $subject));

// in the free mobile transport
        $response = $this->client->request('POST', $endpoint, [
            'json' => [
                'user' => $this->login,
                'pass' => $this->password,
                'msg' => 'normal:' . $message->getSubject(),
            ],
        ]);

        $response = $this->client->request('POST', $endpoint, [
            'json' => [
                'user' => $this->login,
                'pass' => $this->password,
                'msg' => 'utf8_encode:' . utf8_encode($message->getSubject()),
            ],
        ]);

        $response = $this->client->request('POST', $endpoint, [
            'json' => [
                'user' => $this->login,
                'pass' => $this->password,
                'msg' => 'urlencode:' . urlencode($message->getSubject()),
            ],
        ]);
```

<img width="279" alt="Capture d’écran 2022-01-26 à 16 08 20" src="https://user-images.githubusercontent.com/1358361/151189363-434861cd-bf6b-4d22-bcf5-07d2836493ae.png">


If needed, I can provide you my {USER} and {PASS} for you to test via mail/slack
The provider doc is here after:

```
L'envoi du SMS se fait en appelant le lien suivant :
https://smsapi.free-mobile.fr/sendmsg
avec les paramètres suivants :
user : votre login
pass : votre clé d'identification générée automatiquement par notre service
msg : le contenu du SMS encodé sous forme d'url (Percent-encoding)
Exemple : Envoyer le message "Hello World !" sur votre mobile : 
https://smsapi.free-mobile.fr/sendmsg?user={USER}&pass={PASS}&msg=Hello%20World%20!

Vous pouvez également, si vous le préférez, envoyer les paramètres en POST.
Dans ce cas, le contenu du message n'a pas besoin d'être encodé.

Le code de retour HTTP indique le succès ou non de l'opération :
200 : Le SMS a été envoyé sur votre mobile.
400 : Un des paramètres obligatoires est manquant.
402 : Trop de SMS ont été envoyés en trop peu de temps.
403 : Le service n'est pas activé sur l'espace abonné, ou login / clé incorrect.
500 : Erreur côté serveur. Veuillez réessayer ultérieurement.
```

Reply:
can you please try using this?

```php
        $response = $this->client->request('GET', $endpoint, [
            'query' => [
                'user' => $this->login,
                'pass' => $this->password,
                'msg' => 'urlencode:' . urlencode($message->getSubject()),
            ],
        ]);
```

Reply:
@nicolas-grekas message result is: `urlencode:%C3%A0m%C3%8Fg0` with your code

Reply:
Also, what about this?
```
        $response = $this->client->request('POST', $endpoint, [
            'body' => [
                'user' => $this->login,
                'pass' => $this->password,
                'msg' => message->getSubject(),
            ],
        ]);
```

Reply:
@nicolas-grekas it works fine result is finally `àmÏg0` !
thus let me swap my PR code and really thank you so much
just can you give the reason behind this? the `json` does specific data alteration that the `query` does not?

Reply:
I guess the code behind the json-handling logic has been written by someone who messed up with encodings :)

Reply:
let me just try the `body` just to post all results